opentelemetry_collector/receiver/vmagereceiver/vm_age_collector.go

```diff
@@ -99,40 +119,55 @@ func (collector *VMAgeCollector) StopCollection() {
 }
 
 func (collector *VMAgeCollector) makeErrorMetrics() *metricspb.Metric {
-	timeseries := metricgenerator.MakeInt64TimeSeries(1, collector.startTime, time.Now(), collector.labelValues)
+	timeseries := metricgenerator.MakeInt64TimeSeries(1, collector.collectorStartTime, time.Now(), collector.labelValues)
 	return &metricspb.Metric{
-		MetricDescriptor: vmImageErrorMetric,
+		MetricDescriptor: vmImageAgesErrorMetric,
 		Timeseries:       []*metricspb.TimeSeries{timeseries},
 	}
+}
+
+func makeMetrics(metricDescriptor *metricspb.MetricDescriptor, timeseries *metricspb.TimeSeries) []*metricspb.Metric {
+	return []*metricspb.Metric{
+		{
+			MetricDescriptor: metricDescriptor,
+			Timeseries:       []*metricspb.TimeSeries{timeseries},
+		},
+	}
+}
 
+func (collector *VMAgeCollector) export(metrics []*metricspb.Metric, errorKey string) {
+	ctx := context.Background()
+	md := consumerdata.MetricsData{Metrics: metrics}
+	err := collector.consumer.ConsumeMetrics(ctx, internaldata.OCSliceToMetrics([]consumerdata.MetricsData{md}))
+	if err != nil {
+		collector.logger.Error(errorKey, zap.Error(err))
+	}
 }
 
-func (collector *VMAgeCollector) scrapeAndExport() {
-	metrics := make([]*metricspb.Metric, 0, 1)
+func (collector *VMAgeCollector) scrapeAndExportVMImageAge() {
+	var metrics []*metricspb.Metric
 
 	if collector.buildDateError {
-		metrics = append(metrics, collector.makeErrorMetrics())
+		metrics = []*metricspb.Metric{collector.makeErrorMetrics()}
 	} else {
 		imageAge, err := calculateImageAge(collector.parsedBuildDate, time.Now())
 		if err != nil {
-			metrics = append(metrics, collector.makeErrorMetrics())
+			metrics = []*metricspb.Metric{collector.makeErrorMetrics()}
 		} else {
-			timeseries := metricgenerator.MakeDoubleTimeSeries(
-				imageAge, collector.startTime, time.Now(), collector.labelValues)
-			metrics = append(
-				metrics,
-				&metricspb.Metric{
-					MetricDescriptor: vmImageAgeMetric,
-					Timeseries:       []*metricspb.TimeSeries{timeseries},
-				},
-			)
+			timeseries := metricgenerator.MakeDoubleTimeSeries(imageAge, collector.collectorStartTime, time.Now(), collector.labelValues)
+			metrics = makeMetrics(vmImageAgeMetric, timeseries)
 		}
 	}
 
-	ctx := context.Background()
-	md := consumerdata.MetricsData{Metrics: metrics}
-	err := collector.consumer.ConsumeMetrics(ctx, internaldata.OCSliceToMetrics([]consumerdata.MetricsData{md}))
-	if err != nil {
-		collector.logger.Error("Error sending metrics", zap.Error(err))
+	collector.export(metrics, "Error sending VM image age metrics")
+}
+
+func (collector *VMAgeCollector) scrapeAndExportVMReadyTime() {
+	if collector.vmReadyTimeError {
+		return
 	}
+
+	readyTime := int64(collector.parsedVMReadyTime.Sub(collector.parsedVMStartTime) / time.Second)
```

Comment:
This is always going to produce the same result, so it can be done outside the export loop.